src/test/java/com/desiato/puresynth/models/UserControllerTest.java

```diff
@@ -43,6 +45,28 @@ public void createUser_ShouldReturnCreatedUser() throws Exception {
                 .andDo(print());
     }
 
+    @Test
+    public void deleteUser_ShouldDeleteUserAndReturnOk() throws Exception {
+        String uniqueEmail = generateUniqueEmail();
+        String newUserJson = String.format("""
+    {
+        "email": "%s",
+        "password": "password123"
+    }
+    """, uniqueEmail);
+
+        MvcResult result = mockMvc.perform(post("/api/user")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(newUserJson))
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        Long userId = JsonPath.parse(result.getResponse().getContentAsString()).read("$.id", Long.class);
+
+        mockMvc.perform(delete("/api/user/" + userId))
+                .andExpect(status().isOk());
```

Comment:
You should also check, that this user was indeed removed from the DB. You can do that in one of two ways (both are correct):
* do a service call, that a user doesn't exist
* do a `GET` API call to check user doesn't exist.

Reply:
used second approach